docs/data-cookbooks/sns.md

```diff
@@ -0,0 +1,123 @@
+# SNS Notifications in Workflows
+
+On deployment, an sftracker (Step function tracker) [SNS](https://aws.amazon.com/sns) topic is created and used for messages related to the workflow.
+
+Workflows can be configured to send SNS messages containing the Cumulus message throughout the workflow by using the [SF SNS Report lambda function](https://www.npmjs.com/package/@cumulus/sf-sns-report).
+
+## Pre-Deployment Configuration
+
+### Workflow Configuration
+
+The [Hello World Workflow](./hello-world.md) is configured to send an SNS message when starting the workflow and upon workflow completion. This is configured in `workflows.yml`.
+
+```
+HelloWorldWorkflow:
+  Comment: 'Returns Hello World'
+  StartAt: StartStatus
+  States:
+    StartStatus:
+      Type: Task
+      Resource: ${SfSnsReportLambdaFunction.Arn} # This will send a status message at the start of the workflow
+      CumulusConfig:
+        cumulus_message:
+          input: '{$}' # Configuration to send the payload to the SNS Topic
+      Next: HelloWorld
+    HelloWorld:
+      CumulusConfig:
+        buckets: '{$.meta.buckets}'
+        provider: '{$.meta.provider}'
+        collection: '{$.meta.collection}'
+      Type: Task
+      Resource: ${HelloWorldLambdaFunction.Arn}
+      Next: StopStatus
+    StopStatus:
+      Type: Task
+      Resource: ${SfSnsReportLambdaFunction.Arn} # This will send a success status message at the end of the workflow
+      CumulusConfig:
+        sfnEnd: true # Indicates the end of the workflow
+        stack: '{$.meta.stack}'
+        bucket: '{$.meta.buckets.internal.name}'
+        stateMachine: '{$.cumulus_meta.state_machine}'
+        executionName: '{$.cumulus_meta.execution_name}'
+        cumulus_message:
+          input: '{$}' # Configuration to send the payload to the SNS Topic
```

Comment:
Not a problem with the PR, but cleaning up the SNS stuff is a topic for a future architecture meeting.  We send the entire output message.  The API (I think) makes unwritten assumptions about how it's structured in order to pull out collections and granules and all that.

Ideally, we should just have one task "send-sns" that takes a topic ARN and message and does its thing.

Again, not to be addressed here, but this is bringing it up.